### PR TITLE
[ci] Test source_controller

### PR DESCRIPTION
**Pull request by Ana06**
Test `source_controller` methods that are called in the old test suit from `project_controller_tests.rb`. :bowtie:

### CODE REVIEW

**Comment by Ana06** [comment]
[(3) P6: app/controllers/source_controller.rb](https://trello.com/c/HMBD9G3k/275-3-p6-app-controllers-source-controller-rb)

File: src/api/spec/controllers/source_controller_spec.rb [hunk 1]
```diff
@@ -24,4 +31,23 @@
...
   end
+
+  describe "GET #show_project_meta" do
+    before do
+      login user
+      get :show_project_meta, params: { project: project }
+    end
+
+    it { expect(response).to be_success }
+    it {expect(Xmlhash.parse(response.body)["name"]).to eq(project.name)}
```
Comment (bgeuken) [inline src/api/spec/controllers/source_controller_spec.rb]: whitespaces are missing here

File: src/api/spec/controllers/source_controller_spec.rb [hunk 2]
```diff
@@ -24,4 +31,23 @@
...
+  end
+
+  describe "PUT #update_project_config" do
+    before do
+      login user
+      put :update_project_config, params: { project: project, comment: 'Updated by test' }
+    end
+
+    it { expect(response).to be_success }
```
Comment (bgeuken) [inline src/api/spec/controllers/source_controller_spec.rb]: I would also check if the comment got created

File: src/api/spec/controllers/source_controller_spec.rb [hunk 3]
```diff
@@ -1,6 +1,13 @@
 require 'rails_helper'
+# WARNING: If you change owner tests make sure you uncomment this line
+# and start a test backend. Some of the Owner methods
+# require real backend answers for projects/packages.
+# CONFIG['global_write_through'] = true
```
Comment (evanrolfe) [inline src/api/spec/controllers/source_controller_spec.rb]: Does this comment apply to all controller specs? Maybe it should go in the test suite guide instead?
https://github.com/evanrolfe/open-build-service/blob/master/src/api/spec/README.md

Also what are the "owner tests" and "owner methods"?

**Comment by Ana06** [comment]
@evanrolfe 
> Does this comment apply to all controller specs? Maybe it should go in the test suite guide instead?

Not to all of them, only the ones that gave calls to the backend. And yes we should add it to the guide, I would do it in another PR as it is unrelated. :wink:

File: src/api/spec/controllers/source_controller_spec.rb [hunk 4]
```diff
@@ -24,4 +30,25 @@
...
+
+    it { expect(response).to be_success }
+    it { expect(Xmlhash.parse(response.body)["name"]).to eq(project.name) }
+  end
+
+  describe "PUT #update_project_config" do
+    before do
+      login user
+      put :update_project_config, params: { project: project, comment: 'Updated by test' }
+      #project.reload
```
Comment (bgeuken) [inline src/api/spec/controllers/source_controller_spec.rb]: There is debug code left;-)

File: src/api/spec/controllers/source_controller_spec.rb [hunk 5]
```diff
@@ -24,4 +30,25 @@
...
+    end
+
+    it { expect(response).to be_success }
+    it { expect(project.config.to_s).to include('Updated', 'by', 'test') }
```
Comment (bgeuken) [inline src/api/spec/controllers/source_controller_spec.rb]: JFYI There should also be a comment instance that got created for the project

**Comment by bgeuken** [comment]
@Ana06 \*ping\*

**Pull request merged**